src/component/common/Dialogue/Dialogue.tsx

```diff
@@ -68,18 +71,37 @@ const Dialogue: React.FC<IDialogue> = ({
                 <ConditionallyRender
                     condition={Boolean(onClick)}
                     show={
-                        <Button
-                            form={formId}
-                            color="primary"
-                            variant="contained"
-                            onClick={handleClick}
-                            autoFocus={!formId}
-                            disabled={disabledPrimaryButton}
-                            data-test={DIALOGUE_CONFIRM_ID}
-                            type={formId ? 'submit' : 'button'}
-                        >
-                            {primaryButtonText || "Yes, I'm sure"}
-                        </Button>
+                        <ConditionallyRender
```

Comment:
Perhaps we could simply pass an entire PermissionButton in instead of the permission? Otherwise we might have to accept a lot of arguments that are not specific to the dialogue, such as projectId and environment - which will be available in the enclosing scope of the component that is actually using the dialogue component.